src/main/java/net/minecraftforge/common/util/LazyOptional.java

```diff
@@ -23,12 +23,13 @@
 import java.util.Objects;
 import java.util.Optional;
 import java.util.Set;
-import java.util.concurrent.atomic.AtomicReference;
```

Comment:
What's the purpose of moving away from the standard library functions?
The Apache tools are much less useful for this, as we wanted to keep it Atomic, which Apache's Mutable has notable shortcomings with...

Reply:
What use would keeping it atomic have in the new implementation? It would most likely be slower (no idea how much), and the multithreading is correct without it (the lock deals with synchronization).

Reply:
As I said on the issue, keep the atomic and don't make weird random state value things.
Any claims of it being 'slower' need to actually have statistics to back it up.
Currently your doing a 36 liner change for something that could be 2.